src/Parsers.jl

```diff
@@ -170,6 +170,62 @@ Options(;
+abstract type AbstractFloatParseConf{T}    <: AbstractTypeParseConf{T} end
+abstract type AbstractIntParseConf{T}      <: AbstractTypeParseConf{T} end
+abstract type AbstractStringParseConf{T}   <: AbstractTypeParseConf{T} end
+
+struct BoolConf{Bool} <: AbstractTypeParseConf{Bool}
+    refs::Vector{String}
+    trues::Union{Nothing, Vector{PtrLen}}
+    falses::Union{Nothing, Vector{PtrLen}}
+    # TODO: kwargs + validate inputs like Options does
+    BoolConf(::Type{T}=Bool, args...; kwargs...) where {T} = BoolConf{T}(args...; kwargs...)
+    BoolConf{Bool}(options::Options=OPTIONS, refs=options.refs, trues=options.trues, falses=options.falses) = new{Bool}(refs, trues, falses)
+    BoolConf{Bool}(refs, trues, falses) = new{Bool}(refs, trues, falses)
+end
+struct DateTimeConf{T<:Union{DateTime,Date,Time}} <: AbstractDateTimeParseConf{T}
+    dateformat::Union{Nothing, Format}
+    # TODO: kwargs + validate inputs like Options does
+    DateTimeConf(::Type{T}, args...; kwargs...) where {T} = DateTimeConf{T}(args...; kwargs...)
+    DateTimeConf{T}(options::Options=OPTIONS, fmt=options.dateformat) where{T} = new{T}(fmt)
+    DateTimeConf{T}(fmt) where{T} = new{T}(fmt)
+end
+struct FloatConf{T<:Real} <: AbstractFloatParseConf{T}
+    decimal::UInt8
+    groupmark::Union{Nothing,UInt8}
+    # TODO: kwargs + validate inputs like Options does
+    FloatConf(::Type{T}, args...; kwargs...) where {T} = FloatConf{T}(args...; kwargs...)
+    FloatConf{T}(options::Options=OPTIONS, decimal=options.decimal, groupmark=options.groupmark) where {T} = new{T}(decimal, groupmark)
+    FloatConf{T}(decimal, groupmark) where {T} = new{T}(decimal, groupmark)
+end
+struct IntConf{T<:Integer} <: AbstractIntParseConf{T}
+    groupmark::Union{Nothing,UInt8}
+    # TODO: kwargs + validate inputs like Options does
+    IntConf(::Type{T}, args...; kwargs...) where {T} = IntConf{T}(args...; kwargs...)
+    IntConf{T}(options::Options=OPTIONS, groupmark=options.groupmark) where {T} = new{T}(groupmark)
+    IntConf{T}(groupmark) where {T} = new{T}(groupmark)
+end
+struct StringConf{T<:Union{AbstractString,Char,Symbol}} <: AbstractStringParseConf{T}
+    stripwhitespace::Bool
+    stripquoted::Bool
+    # TODO: kwargs + validate inputs like Options does
+    StringConf(::Type{T}, args...; kwargs...) where {T} = StringConf{T}(args...; kwargs...)
+    StringConf{T}(options::Options=OPTIONS, stripwhitespace=options.stripwhitespace, stripquoted=options.stripquoted) where {T} = new{T}(stripwhitespace, stripquoted)
+    StringConf{T}(stripwhitespace, stripquoted) where {T} = new{T}(stripwhitespace, stripquoted)
+end
+
+struct UnknownTypeConf{T} <: AbstractTypeParseConf{T} end
+
+conf(::Type{T},    opt_or_conf) where {T<:Integer}                           = IntConf{T}(opt_or_conf.groupmark)
+conf(::Type{T},    opt_or_conf) where {T<:Real}                              = FloatConf{T}(opt_or_conf.decimal, opt_or_conf.groupmark)
```

Comment:
Technically, our float parsing only supports `Parsers.SupportedFloats`, so I think we'd want to be a bit tighter here than `Real`?

Reply:
Scratch that, I forgot we _do_ support all `Real` and just parse `Float64` and convert from that.

Reply:
Yeah exactly